packages/truffle-workflow-compile/index.js

```diff
@@ -72,29 +72,37 @@ const Contracts = {
   // network_id: network id to link saved contract artifacts.
   // quiet: Boolean. Suppress output. Defaults to false.
   // strict: Boolean. Return compiler warnings as errors. Defaults to false.
-  compile: callbackify(async function(options) {
-    const config = prepareConfig(options);
-
-    const compilers = config.compiler
-      ? [config.compiler]
-      : Object.keys(config.compilers);
-
-    this.reportCompilationStarted(options);
-
-    const compilations = await this.compileSources(config, compilers);
-
-    const numberOfCompiledContracts = compilations.reduce(
-      (number, compilation) => {
-        return number + Object.keys(compilation.contracts).length;
-      },
-      0
-    );
-
-    if (numberOfCompiledContracts === 0) this.reportNothingToCompile(options);
-
-    this.reportCompilationFinished(options, config);
-    return await this.collectCompilations(compilations);
-  }),
+  compile: async function(options, callback) {
```

Comment:
I think in order to make this polymorphic, we can't make it `async`, because it will return a promise no matter what. Not sure though, maybe this is fine as a best practice?

Reply:
Ah yes I see what you mean. Let me give it a mull-over.

Reply:
After working this method out in an alternate way, I'm going to go ahead and claim that it is not a problem the way it currently is in this PR. It is not too bad so if you feel strongly about changing it, then I can go ahead and make the swap no problem. Otherwise, I think it is a bit impractical write it and make sure it doesn't return anything in the case that a callback is passed.